### PR TITLE
RegEx Matching (Independent Publisher)

### PR DESCRIPTION
**Pull request by happyume**
### Images
Screenshot_RegEx_CreateFlow_001
Screenshot_RegEx_CreateFlow_002
Screenshot_RegEx_ActionList
Screenshot_RegEx_SuccessResult_001
Screenshot_RegEx_SuccessResult_002
Screenshot_RegEx_CreateFlow_003
Screenshot_RegEx_SuccessResult_003
Screenshot_RegEx_TestCLI

---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by v-EgorKozhadei** [comment]
Hi @happyume!

Thank you for submitting your submission!

Your PR is missing [screenshots from Test operations section](https://github.com/microsoft/PowerPlatformConnectors#independent-publisher-connectors-1), can you please add them?
You can refer to other open Independent Published PRs as an example.

Please let me know if you have any questions!

**Comment by happyume** [comment]
Hi [v-EgorKozhadei](https://github.com/v-EgorKozhadei),
I have now uploaded all the screenshots as required. 
Please can you review. 
Thanks

**Comment by natalie-pienkowska** [comment]
Hi @happyume thanks so much for your contribution :) If you've submitted all the connector artifacts and feel ready for certification, please remove "Proposal" from your pull request title.

**Comment by happyume** [comment]
Hi @natalie-pienkowska - The title is now updated to remove the word proposal. Thanks for the reminder :)

**Comment by v-EgorKozhadei** [comment]
Hi again @happyume! Can you please remove the screenshots from the changed files and add them as images to the PR description? Screenshots from Test operations section are usually added here, not to the connector files. Sorry that I didn't mention it in my previous comments!

File: independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/readme.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## RegEx Matching
+This connector extends Power Automate capabilities for performing tests whether a text string matches a pattern. There are pre-built patterns and also an option for user to provide their own regular expressions (regex) for matching the text string. 
+
+## Publisher
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/readme.md:4]: @happyume please format the publisher, it should look like `Publisher: Publisher's Name`.

File: independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/readme.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+| `valid GUID` | Returns `true` if the text string is in a valid GUID format |
+| `valid SSN` | Returns `true` if the text string is in United States Social Security Number format |
+| `valid Credit Card` | Returns `true` if the text string is in credit card number format |
+| `valid DATETIME` | Returns `true` if the text string is in YYYYMMDDHHMMSS format |
+| `contains Digit` | Returns `true` if the text string contains a digit (0-9) anywhere in the input |
+| `startsWith` | Returns `true` if the text string starts with the given keyword |
+| `endsWith` | Returns `true` if the text string ends with the given keyword |
+| `match pattern` | Returns `true` if the text string matches the custom regex pattern |
+
+## Limitations and Exclusions
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/readme.md:23]: @happyume please format the readme.md file according to [the template](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/readme.md). All the required headings should be present, they also should be placed in the correct order, like in the template.

File: independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+      "propertyName":"Website",
+      "propertyValue":"https://sa018.azurewebsites.net"
+    },
+    {
+      "propertyName":"Privacy Policy",
+      "propertyValue":"https://sa018.azurewebsites.net"
+    },
+    {
+      "propertyName":"Categories",
+      "propertyValue":" Content and Files"
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/apiDefinition.swagger.json:29]: @happyume please remove the whitespace from the string.

File: independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+    }
+  ],
+  "consumes": [],
+  "produces": [],
+  "paths": {
+    "/api/validEmail/": {
+      "get": {
+        "responses": {
+          "200": {
+            "description": "default",
```
Comment (v-EgorKozhadei) [inline independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/apiDefinition.swagger.json:39]: @happyume please add a more detailed description for your 2xx code for all operations, `default` description doesn't fit here. You can refer to [the documentation](https://docs.microsoft.com/en-us/connectors/custom-connectors/coding-standards#responses) for an example.

**Review by v-EgorKozhadei — COMMENTED** [review]
Hi @happyume! Please check the comments after the review, thank you!

File: independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/readme.md (hunk 1 above)
Comment (happyume) [reply independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/readme.md]: This is now updated.

File: independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/apiDefinition.swagger.json (hunk 3 above)
Comment (happyume) [reply independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/apiDefinition.swagger.json]: This is now updated

File: independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/apiDefinition.swagger.json (hunk 4 above)
Comment (happyume) [reply independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/apiDefinition.swagger.json:39]: This is now updated

File: independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/readme.md (hunk 2 above)
Comment (happyume) [reply independent-publisher-connectors/Proposal - RegEx Matching (Independent Publisher)/readme.md]: This is now updated

**Comment by happyume** [comment]
Hi @v-EgorKozhadei  - Thank-you for advising me of the corrections to be done. I have now updated as per your comments. Please can you let me know if any further changes are needed.

**Comment by v-EgorKozhadei** [comment]
Hi @happyume! Thank you very much for all the updates!
Can you please add the screenshots to the PR description and remove it from the folder with connector files?
The files changed tab contains 11 files, we need only 3: apiProperties.json, apiDefinition.swagger.json and readme.md.

You can [refer to this PR](https://github.com/microsoft/PowerPlatformConnectors/pull/1685) for an example. The screenshots in this one are in PR's description and the connector folder contains only 3 files.

Please let me know if you have any questions or need help!

**Comment by happyume** [comment]
Hi @v-EgorKozhadei  - I have now deleted the sub-directory images (with 8 screenshots) and loaded them in this Pull Request description (first post at the top).

**Comment by v-EgorKozhadei** [comment]
Hi @happyume! Thanks a lot for all the updates!

I'd like to ask you for one more fix before we can proceed to the next step.
Can you please rename the connector folder from `Proposal - RegEx Matching (Independent Publisher)` to just `RegEx Matching`?

**Comment by happyume** [comment]
Hi @v-EgorKozhadei  - I have created a different directory (without the word Proposal) with the reviewed files and deleted the directory which had the word Proposal

**Comment by v-EgorKozhadei** [comment]
Hi @happyume! Thanks a lot for the changes! Can you please also remove the `(Independent Publisher)` part as well? So that the folder name will be just `RegEx Matching`.

**Comment by happyume** [comment]
Hi @v-EgorKozhadei - Apologies, I misunderstood. This is now fixed.

**Comment by v-EgorKozhadei** [comment]
[[certify-connector]]

**Comment by v-EgorKozhadei** [comment]
Hi @happyume, 

Congratulations, your connector is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks.  

Please let us know if you have any questions. 

Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
Hello @happyume,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by sriyen-msft** [comment]
@happyume Congratulations! The deployment of your connector and documentation page is now complete, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program, Thanks.

**Review by sriyen-msft — APPROVED** [review]

**Pull request merged**